ExtraEnemyCustomization/CustomAbilities/DrainStamina/DrainStaminaManager.cs

```diff
@@ -0,0 +1,32 @@
+﻿using EECustom.Attributes;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace EECustom.CustomAbilities.DrainStamina
+{
+    [CallConstructorOnLoad]
+    public static class DrainStaminaManager
+    {
+        internal static DrainStaminaSync Sync { get; private set; } = new();
+
+        static DrainStaminaManager()
+        {
+            Sync.Setup();
+        }
+
+        public static void DoDrain(DrainStaminaData data)
+        {
+            Sync.Send(data);
+        }
+    }
+
+    public struct DrainStaminaData
+    {
+        public int playerSlot;
```

Comment:
Probably should use player lookup instead of player slot.
Are you trying to add support for bots?

Reply:
I think the pPlayer struct has references for bots? I could be wrong though